src/interactivity/base/RemoteConsoleControl.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "RemoteConsoleControl.hpp"
+
+#include "../../inc/HostSignals.hpp"
+
+using namespace Microsoft::Console::Interactivity;
+
+RemoteConsoleControl::RemoteConsoleControl(HANDLE signalPipe) :
+    _pipe{ signalPipe }
+{
+}
+
+#pragma region IConsoleControl Members
+
+template<typename T>
+[[nodiscard]] NTSTATUS _SendTypedPacket(HANDLE pipe, ::Microsoft::Console::HostSignals signalCode, T& payload)
+{
+    struct HostSignalPacket
+    {
+        ::Microsoft::Console::HostSignals code;
+        T data;
+    };

```

Comment:
This may not work the way you'd like, unfortunately. Intra-member padding will be introduced for alignment purposes s.t. you'll get `[code 0 0 0 data1 data2 data3]` on the wire

Reply:
pack should do it.